pkg/upgrade/key_pair.go

```diff
@@ -8,42 +8,55 @@ import (
 	"strings"
 
 	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
-	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/services/certificates"
-	clusterapiv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
-// KeyPair contains a cert and key.
-type KeyPair struct {
+// KubeconfigSecretKey is the key where the kubeconfig is stored in the secret.
+const KubeconfigSecretKey = "kubeconfig"
+
+// keyPair contains a cert and key.
+type keyPair struct {
 	Cert []byte `json:"cert"`
 	Key  []byte `json:"key"`
 }
 
-type keyPairGetter struct {
-	secretClient corev1client.SecretsGetter
+// secrets has already been scoped to namespace, so, .Secrets("my-namespace") returns one of these.
+type secrets interface {
+	Get(string, metav1.GetOptions) (*v1.Secret, error)
```

Comment:
If you wouldn't mind, please name the parameters

Reply:
yeah, no worries, that definitely improves readability